### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by kentsanggds**
Files renamed and tests placed in tests/functional instead of tests/integration as thats probably a better fit

Also finally got the tests passing after upgrading phantomJS on the Jenkins slave.

### CODE REVIEW

**Comment by andyhd** [comment]
There are a lot of commits in here that are unrelated to implementing integration tests - could you clean this PR up a bit?

File: paas-deploy.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# Note: the actual name of the environment variable is determined
+# by what you enter into the Credentials Binding Plugin
+cf auth "$CF_USER" "$CF_PASSWORD"
+
+cf target -o myorg -s myspace
+cf push ags-gateway-aws
+
+# Destroy token
+cf logout
```
Comment (andyhd) [inline paas-deploy.sh]: this commit should be in a separate PR

File: app/templates/views/auth/confirm_email.html [hunk 2]
```diff
@@ -47,6 +47,9 @@ <h1 class="heading-large">Do you know your work email?</h1>
     <div class="form-group">
       <button class="button">Continue</button>
     </div>
+    <script type="text/javascript">
+        document.querySelector('#email_known-0').checked = true
+    </script>
```
Comment (andyhd) [inline app/templates/views/auth/confirm_email.html]: Is this meant to be committed?

File: run-tests [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 
 import argparse
+from lib.govuk_assets import install_govuk_assets
 
 
 BASE_TEST_RUNNER_ARGS = [
     '--eradicate',
     '--flake8',
     '--spec',
+    '--junitxml=log/ags_gateway.log',
```
Comment (andyhd) [inline run-tests]: This belongs in a separate PR

File: tests/functional/test_functional.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from flask import url_for
+
+
+class TestFunctional(object):
```
Comment (andyhd) [inline tests/functional/test_functional.py]: This could be better named

File: tests/functional/test_functional.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    def test_can_show_email_confirmation(self, live_server, browser):
+        browser.visit(url_for('main.request_email_address', _external=True))
+
+        assert ('GOV.UK - The best place to find government services and '
+                'information') in browser.title
```
Comment (andyhd) [inline tests/functional/test_functional.py]: This breaks if the copy changes, and do we need to make this assertion anyway?

File: tests/functional/test_functional.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+        assert browser.is_text_present('Do you know your work email?')
```
Comment (andyhd) [inline tests/functional/test_functional.py]: Don't make assertions on the copy - the test will break if the copy changes.

File: tests/functional/test_functional.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    def test_can_go_from_email_confirm_to_department_confirmation(
+            self, live_server, browser):
+        browser.visit(url_for('main.request_email_address', _external=True))
+
+        browser.choose('email_known', 'yes')
+
+        browser.fill('email_address',
+                     'some.one@digital.cabinet-office.gov.uk')
```
Comment (andyhd) [inline tests/functional/test_functional.py]: Refactor the email address into a variable, eg: `email = 'some.one@digital...'`

File: tests/functional/test_functional.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+        browser.find_by_xpath(
+            '//*[@id="content"]/form/div[3]/button').click()
```
Comment (andyhd) [inline tests/functional/test_functional.py]: Prefer CSS to Xpath (just because more people are familiar with CSS selectors)

File: tests/functional/test_functional.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+        assert browser.is_text_present(
+            'Email address you provided')
```
Comment (andyhd) [inline tests/functional/test_functional.py]: This assertion is brittle and unnecessary

File: tests/functional/test_functional.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        assert browser.is_text_present(
+            'some.one@digital.cabinet-office.gov.uk')
```
Comment (andyhd) [inline tests/functional/test_functional.py]: Better to assert that a specific element contains the expected text - if the form submit failed, this might pass due to the email address being in the form input.

File: tests/functional/test_functional.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+        assert browser.is_text_present(
+            'Do you work for the Government Digital Service?')
```
Comment (andyhd) [inline tests/functional/test_functional.py]: Is this assertion needed?

File: app/templates/views/auth/confirm_email.html (hunk 2 above)
Comment (kentsanggds) [reply app/templates/views/auth/confirm_email.html]: Sorry, was part of testing, will take it out

File: run-tests (hunk 3 above)
Comment (kentsanggds) [reply run-tests]: Will change from PR

File: tests/functional/test_functional.py (hunk 4 above)
Comment (kentsanggds) [reply tests/functional/test_functional.py]: Will rename as Test High Level User Flows

File: tests/functional/test_functional.py (hunk 5 above)
Comment (kentsanggds) [reply tests/functional/test_functional.py]: This was just an initial test to confirm that selenium is working, can be removed as doesn't do anything except go to the page

File: tests/functional/test_functional.py (hunk 6 above)
Comment (kentsanggds) [reply tests/functional/test_functional.py]: OK test will be removed

File: tests/functional/test_functional.py (hunk 7 above)
Comment (kentsanggds) [reply tests/functional/test_functional.py]: OK

File: tests/functional/test_functional.py (hunk 8 above)
Comment (kentsanggds) [reply tests/functional/test_functional.py]: Was not aware of that as I normally use xpath, will update

File: tests/functional/test_functional.py (hunk 9 above)
Comment (kentsanggds) [reply tests/functional/test_functional.py]: OK, will revise assertions

File: tests/functional/test_functional.py (hunk 10 above)
Comment (kentsanggds) [reply tests/functional/test_functional.py]: OK

File: tests/functional/test_functional.py (hunk 11 above)
Comment (kentsanggds) [reply tests/functional/test_functional.py]: Thought it would be useful to ensure that the correct department is picked up from the email

Comment (andyhd) [reply tests/functional/test_functional.py]: Ah yes - ok, this is fine then. But maybe check for "Government Digital Service" appearing in a specific element rather than anywhere on the page.

Comment (kentsanggds) [reply tests/functional/test_functional.py]: OK
